### PR TITLE
[automate-1877] Remove IAM v2.0 specific code from authz-service

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

Remove as much IAM V2.0 specific stuff from authz-service as possible. Removed the V2 version of the `IsAuthorized` function as well. 

### :white_check_mark: Checklist

- [ ] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: components/event-gateway/pkg/nats/authenticator.go [hunk 1]
```diff
@@ -283,19 +283,22 @@ func (a *automateAuthenticator) checkToken(token string) error {
 }
 
 func (a *automateAuthenticator) tryAuthzV2(ctx context.Context, subjects []string) (err error, shouldTryV1 bool) {
-	resp, err := a.authzClient.IsAuthorized(ctx, &authz.IsAuthorizedReq{
+	resp, err := a.authzClient.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
```
Comment (tylercloke) [inline components/event-gateway/pkg/nats/authenticator.go]: The fact that this code exists at all is a giant smell to me. It wasn't working at all with IAM V2.1 so this at least pushes it in the right direction.

File: components/authz-service/engine/conformance/conformance_test.go [hunk 2]
```diff
@@ -671,22 +671,6 @@ func setupV1(t testing.TB) (context.Context, map[string]engine.Engine) {
...
 
-func setupV2(t testing.TB) (context.Context, map[string]engine.Engine) {
-	ctx, engines := setup(t)
-	if o, ok := engines["opa"]; ok {
-		o.V2SetPolicies(ctx, map[string]interface{}{}, map[string]interface{}{})
-	}
-	return ctx, engines
-}
-
-func setupV2p1(t testing.TB) (context.Context, map[string]engine.Engine) {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/engine/conformance/conformance_test.go]: I moved it to the file that was actually using it.

File: components/authz-service/engine/opa/opa_internal_test.go [hunk 3]
```diff
@@ -402,7 +402,7 @@ func BenchmarkV2FilterAuthorizedPairsRealWorldExample(b *testing.B) {
 			var resp []engine.Pair
 			var err error
 			for n := 0; n < b.N; n++ {
-				resp, err = s.V2FilterAuthorizedPairs(ctx, append([]string{"user:local:test@example.com"}, randomTeams(count)...), pairs, true)
+				resp, err = s.V2FilterAuthorizedPairs(ctx, append([]string{"user:local:test@example.com"}, randomTeams(count)...), pairs)
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/engine/opa/opa_internal_test.go]: I'll dig a bit deeper here.

[the remark this answers is not in the document]
Comment (bcmdarroch) [reply components/authz-service/engine/opa/opa_internal_test.go]: ok so we're still using FilterAuthorizedPairs for v2 introspection. but I'm concerned we don't seem to be using the projects in the rego for that: https://github.com/chef/automate/blob/692f5a5ee87b825f6b7525f05741a5ef7d114057/components/authz-service/engine/opa/policy/introspection_v2.rego#L29

i can test out that assumption later today

**Pull request merged**